### ISSUE TITLE
Agent uses hard-coded list of web browsers

### ISSUE THREAD

**Issue by hagabaka** [andrew-bibb/cmst#86]
I just noticed that connman and CMST are able to prompt me to open a browser to log in a public wifi hotspot when necessary. That's very cool. However, CMST only provides a small list of browsers (firefox, opera, luakit, and lynx), and a box to type in my own browser, and it doesn't remember the browser I type in, so I have to do it every time.

I can see two ways to improve this user interface:

1. Include `xdg-open` in the hard-coded list. It should automatically follow settings from the BROWSER environment variable and/or desktop environment.
2. Remember the browser command entered by the user and choose the same option the next time the agent is open.

**Comment by andrew-bibb** [comment]
I can do both, thank you.

I had completely forgotten about that code.  I put it in mainly becase the browser option was in the Connman API.  Maybe a stupid question; but did you try it out, and if you did, did it work?  I've never been to a place (usually hotels) where I've both needed it, and had CMST with me at the time.   The list of browsers are things I was using when I did the coding, and I do recall it was very short, maybe Luakit and Opera, perhaps Lynx.  Two of those I don't even have on my machine anymore.  

Anyway, good observation.  We'll get them implemented.

**Comment by hagabaka** [comment]
Yes the "other browser" option works. I actually worked for a few hours to fix this and am about to create a pull request. It makes somewhat bigger changes in the UI and code though. We can discuss in the pull request.

**Comment by andrew-bibb** [comment]
I'm reopening this one as I've yet to find a way to reopen a pull request thread after I do a pull.  Probably a way, just I've not found it.

I just uploaded to the **devel** branch code to save the last browser selected and present it as the default next time you enter the dialog. 

On the wireless tab I've also got a temporary button called "browser test".  This is linked to the browser select dialog to make it easier to see how it works.  Since this is in the **devel** branch we can have neat little toys like this without worrying about them being used.

I've also added in the "qupzilla" browser to the list (what I currently use).  We probably should go through and try to find any others out there and add them in.
